text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+* fetch all versions of a dataset
+* fetch a specific historical version of a dataset
+* re-fetch once a dataset update has been published
+
+The second problem comes from poor metaphor choices. Moving dataset histories around with "publish" and "add" doesn't map properly to the terms used. "publish" is putting versions of a dataset on a remote peer, and it's really only a "publish" if that peer elects to broadcast the existince of that dataset to other peers, and misses the main thing publish does: place a dataset version on a remote.
+
+We should remove `add` and `publish`, and replace them with two new commands:
+* push: upload datasets to qri peers
+* pull: download datasets from the qri network
+
+Push & pull improve on `add/publish` because they're clear metaphors based on physical verbs. push & pull both describe _who_ is initiating the action (you) and the _directionality_ of the action. Clear defaults, and flags that make behaviour tunable will make a welcome change.
+
+We're also part-way through a transition to renaming "add" to "clone". We're stalled on this transition because our concept of "clone" doesn't presently match what happens within git: `git-clone` both downloads a repo _and checks out the repo into a target directory_. This RFC proposes dropping `clone` altogether, instead folding that functionality into `checkout` by having checkout _automatically pull_ when a dataset isn't local.
+
+Finally, the switch to "greedy pushing" means we need to deal with what happens when a receiver can't handle the entire request. This RFC defines _retention strategies_ as a term for dealing with pushing around more data than a peer is willing to hold, and defines a default retention strategy of `none` to clarify what will happen when a peer tries to push too much data in the short-term.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+I'm going to focus on writing help text that explains the proposed commands. The help text should describe the commands well enough to understand what this RFC is proposing, and define the set of flags that adjust their behaviour.
+
+### push: upload datasets to qri peers
+```
+$ qri push --help
+Push sends datasets to a remote qri node. A push updates the dataset log on the
+remote and sends as many versions as the remote is willing to accept. A remote
+peer determines it's own rules on which versions to accept, but by default
+favour the latest versions of a dataset.
+
+If no remote is specified, qri pushes to the registry. But pushes 
+can just as easily send to another peer that is configured to accept datasets 
+from you. For details on  setting peers for dataset exchange read this doc:
+https://qri.io/docs/hosting-datasets
+
+Usage:
+  qri push [REMOTE] DATASET [DATASET...] [flags]
+
+Aliases:
+  publish
+
+Examples:
+  # push a dataset to the registry
+  $ qri push me/dataset
+
+  # push a specific version of a dataset to the registry:
```

Comment:
assuming this means it *only* pushes the specific version of the dataset to the registry. If so, adding an 'only' in there will clarify even more:
"push only this specific version of a dataset to the registry:"

Q:
Can you use `-- revisions' in conjunction with a version? 
`qri push me/dataset@/ipfs/QmHashOfVersion --revisions 2`
would push version `QmHashOfVersion` as well as the immediately preceding version